crates/hyperdrive-math/src/base.rs

```diff
@@ -1,18 +1,15 @@
 use ethers::types::U256;
-use eyre::{eyre, Result};
+use eyre::{bail, eyre, Result};
 use fixedpointmath::{fixed, FixedPoint};
 
 use crate::State;
 
 impl State {
     /// Calculates the number of share reserves that are not reserved by open
     /// positions.
-    pub fn calculate_idle_share_reserves(&self) -> FixedPoint {
-        let long_exposure = self.long_exposure().div_up(self.vault_share_price());
-        match self.share_reserves() > long_exposure + self.minimum_share_reserves() {
-            true => self.share_reserves() - long_exposure - self.minimum_share_reserves(),
-            false => fixed!(0),
-        }
+    // FIXME: This function is redundant with `calculate_solvency`.
```

Comment:
I agree that we should only have one function, but I think `calculate_idle_share_reserves` is more clear than `calculate_solvency`. The former is more explicit in what is being computed. 

If we allowed negative idle share reserves, then we could also implement a `check_solvency` function that returns a bool based on the sign of `calculate_idle_share_reserves`. Or we do not allow negative idle share reserves and leave it to the developer to know that this is the function to run to check & calculate solvency.